[WIP] Orb Concepts page rewrite (#4540)
circleci/circleci-docs: jekyll/_cci2/orb-concepts.md

### Commands
Commands contain one or more [steps]() in which [parameters]() can be used to modify behavior. Commands are the logic of orbs and are responsible for executing steps such as [checking out code](https://circleci.com/docs/2.0/configuration-reference/#checkout), or running shell code, for example running BASH or running CLI tools.

All orbs are open, meaning that anyone can use them and see their source.
For more information see our guide to [Authoring Reusable Commands]({{site.baseurl}}/2.0/reusing-config/#authoring-reusable-commands).

### Design Methodology
As an example, the AWS S3 orb includes a _command_ to copy a file or object to a new location. If your AWS authentication details are pre-stored as environment variables the syntax to use this command in your config is simply:

Before using orbs, you may find it helpful to understand the various design decisions and methodologies that were used when these orbs were designed. Orbs were designed with the following considerations:

> * Orbs are transparent - If you can execute an orb, you and anyone else can view the source of that orb.

Reviewer: Is this list no longer correct, @KyleTryon?

Author: The Design Methodology section? This seemed to cover some redundant topics, such as semver and immutability (although maybe a quick note on immutability do that). Same with the piece on organization admins, that is covered elsewhere and should probably be touched on again in the development page but seemed redundant here.

will send over a small change for the immutability piece

Author: Expanded on the orb versions section, I think this addresses this.

Reviewer: Brilliant, thanks!

      - aws-s3/copy:
          from: bucket/build_asset.txt
          to: 's3://my-s3-bucket-name'

... # workflows , other jobs etc.
```

### Executors

Executors define the environment in which commands are run. CircleCI provides multiple [execution platforms]({{site.baseurl}}/2.0/configuration-reference/#docker--machine--macos--windows-executor) including:
  - Docker
  - macOS
  - Windows
  - Machine (Linux VM)

In the [Node orb](https://circleci.com/orbs/registry/orb/circleci/node), for example, a parameterized Docker-based executor is provided, through which users can set the Docker tag. This provides a simple way to test applications against any version of Node.js when used with the Node orb's [test job](https://circleci.com/orbs/registry/orb/circleci/node#usage-run_matrix_testing).

```yaml
docker:
  - image: '<org>/<image>:<<parameters.tag>>'
parameters:
  tag:
    default: '13.11'
    type: string
```

For more information, see the guide to [Authoring Reusable Executors]({{site.baseurl}}/2.0/reusing-config/#authoring-reusable-executors) and the registry page for the [Node Orb](https://circleci.com/orbs/registry/orb/circleci/node#executors-default).

> ### Jobs

Reviewer: Need to add in an example here

Author: @rosieyohannan I will add it here but I was thinking we could use the same example we are using for the orb intro page.

Reviewer: Oh yeah good idea

Jobs define a collection of [steps](https://circleci.com/docs/2.0/configuration-reference/#steps) to be run within a given executor. [Workflows]() are then used to orchestrate one or many jobs.

_[See: Authoring Reusable Jobs]({{site.baseurl}}/2.0/reusing-config/#authoring-parameterized-jobs)._
_[See Example: Using Node Test Job](https://circleci.com/orbs/registry/orb/circleci/node#usage-run_matrix_testing)._

## Namespaces

> A _namespace_ is a unique identifier claimed by a user or organization to group a set of orbs by author. Each user and organization may claim _one_ unique and immutable namespace name. Each namespace my contain main uniquely named orbs.

Reviewer: ```suggestion
A _namespace_ is a unique identifier claimed by a user or organization to group a set of orbs by author. Each user or organization can claim _one_ unique and immutable namespace. Each namespace can contain many uniquely named orbs.
```

For example, the `circleci/rails` orb may coexist in the registry with an orb called `<username>/rails` because they are in separate namespaces.

> Organizations are, by default, limited to claiming only one namespace. This policy is designed to limit name-squatting and namespace noise. If you are in need of changing your namespace, please contact your account team at CircleCI.

Reviewer: ```suggestion
Organizations are, by default, limited to claiming only one namespace. This policy is designed to limit name-squatting and namespace noise. If you need to change your namespace, please contact your account team at CircleCI.
```

> By default, any created namespaces will appear as a "community" namespace on the [Orb Registry](https://circleci.com/orbs/registry/).

Reviewer: ```suggestion
By default, created namespaces appear as a "community" namespace in the [Orb Registry](https://circleci.com/orbs/registry/).
```
